### PR TITLE
FOGL-2083: initial implementation of NotificationQueue

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-2083: initial implementation of NotificationQueue

### CODE REVIEW

File: include/notification_queue.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ */
+
+#include <logger.h>
+#include <queue>
+#include <condition_variable>
+#include <rule_plugin.h>
+#include <delivery_plugin.h>
+#include <reading_set.h>
+
+class NotificationQueueElement
```
Comment (MarkRiddoch) [inline include/notification_queue.h]: Doxygen comment please

File: include/notification_queue.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		const std::string&	getAssetName() { return m_assetName; };
+		ReadingSet*		getAssetData() { return m_readings; };
+
+	private:
+		std::string		m_assetName;
+		ReadingSet*		m_readings;
+};
+
+/**
+ * The NotificationQueue class.
```
Comment (MarkRiddoch) [inline include/notification_queue.h]: Could you add some description please

File: include/notification_queue.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ */
+class NotificationQueue
+{
+	public:
+		NotificationQueue(const std::string& serviceName);
```
Comment (MarkRiddoch) [inline include/notification_queue.h]: If it is a singleton we normally make the constructor private

File: notification_queue.cpp [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+ * @param    notificationName	NotificationService name
+ *�
+ */
+NotificationQueue::NotificationQueue(const string& notificationName) :
+				     m_name(notificationName)
+{
+	// Set running
+	m_running = true;
+	// Set instance
+        m_instance = this;
```
Comment (MarkRiddoch) [inline notification_queue.cpp]: formatting

File: include/notification_queue.h (hunk 3 above)
Comment (pintomax) [reply include/notification_queue.h]: Yes. Other classes have similar public method.

I'll review this in next commits

File: notification_queue.cpp (hunk 4 above)
Comment (pintomax) [reply notification_queue.cpp]: Done

**Pull request merged**